openverse-api/catalog/api/models/audio.py

```diff
@@ -0,0 +1,141 @@
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+from uuslug import uuslug
+
+import catalog.api.controllers.search_controller as search_controller
+from catalog.api.models import OpenLedgerModel
+from catalog.api.models.media import (
+    AbstractMedia,
+    AbstractMediaReport,
+    AbstractDeletedMedia,
+    AbstractMatureMedia,
+    AbstractMediaList,
+)
+from catalog.api.models.mixins import (
+    IdentifierMixin,
+    MediaMixin,
+    FileMixin,
+)
+
+
+class AudioSet(IdentifierMixin, MediaMixin, FileMixin, OpenLedgerModel):
+    """
+    This is an ordered collection of audio files, such as a podcast series or
+    an album. Not to be confused with AudioList which is a many-to-many
+    collection of audio files, like a playlist or favourites library.
+
+    The FileMixin inherited by this model refers not to audio but album art.
+    """
+
+    pass
+
+
+class Audio(AbstractMedia):
+    set = models.ForeignKey(
+        help_text='Reference to set of which this track is a part.',
+        to=AudioSet,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+    )
+    set_position = models.IntegerField(
+        blank=True,
+        null=True,
+        help_text='Ordering of the audio in the set.'
+    )
+
+    genre = models.CharField(
+        max_length=80,
+        blank=True,
+        null=True,
+        db_index=True,
+        help_text='',
+    )
+    type = models.CharField(
+        max_length=80,
+        blank=True,
+        null=True,
+        db_index=True,
+        help_text='',
+    )
+
+    duration = models.IntegerField(
```

Comment:
The sound effects in freesound have float-type duration, because some of them are less than one second in length. I'm not sure what's the best way to represent that, though. Milliseconds, maybe?

Reply:
Less than one second? That's too short. Storing them in milliseconds will just add three 0s to all other sounds which are more than 1s long. But I can't think of any alternative approaches so will have to go with that.

Reply:
Tens of millisecond is also an option :)
[Some sounds](https://freesound.org/browse/tags/oh/) seem to be 0 seconds in duration on the search page, but turn out to be 0.xxx seconds when you look at the detail page.

Reply:
Changed the default unit to milliseconds in 9da5519.